### ISSUE TITLE
NVDA installer does not start if special characters are in the installer path

### ISSUE THREAD

**Issue by seanbudd** [nvaccess/nvda#13270]
See also: #9134

### Steps to reproduce:

1. Create a new folder named: `Testé ç ş ğ`
   - copy and paste to preserve characters, there are 4 special characters in this string
1. Place a copy of a NVDA installer (eg 2021.3.1) in the folder named "`Testé ç ş ğ`"
1. Run the installer from the folder
1. Observe popup dialog "NSIS Error" "Error launching installer" 
1. Note that no nvda log is created, this is an early failure.

### Actual behavior:

NVDA installer does not start if special characters are in the installer path

### Expected behavior:

NVDA installer should start if special characters are in the installer path

#### NVDA installed/portable/running from source:

Installer

#### NVDA version:

2021.1 , 2021.1.3, alpha-24501,b45926cd

#### Windows version:

Windows 11 (10.0.22000 Build 22000)

**Comment by ruifontes** [comment]
Tried with NVDA 2021.3.1 and nvda_alpha-24501,b45926cdin a folder named C:\Users\Utilizador\OneDrive - Tiflotecnia, Lda\Transferências and both have installed without problems.
Windows 11 21H2 OS Build 22000.434)

**Comment by seanbudd** [comment]
Would you be able to test with the specific folder name `Testé ç ş ğ` - it is probably a different character causing this problem

**Comment by ruifontes** [comment]
Still without problem...
Folder name:
"C:\Users\Utilizador\OneDrive - Tiflotecnia, Lda\Testé ç s g"

**Comment by seanbudd** [comment]
@ruifontes, This was possibly missed by your synthesiser, both "s" and "g" have accents in the string I was using. I've updated the testing steps to be clearer.

**Comment by ruifontes** [comment]
Copied and pasted the text "Testé ç ş ğ" to the name of a folder and still no problem.

**Comment by seanbudd** [comment]
> Copied and pasted the text "Testé ç ş ğ" to the name of a folder and still no problem.

Strange, perhaps it is an issue with my build of Windows

**Comment by seanbudd** [comment]
To investigate further we need :
- to test a sample NSIS app (and log an issue upstream)
- log the start up NSIS process in detail (if it's an NVDA issue)

**Comment by Brian1Gaff** [comment]
Have we not been here before some years ago, something to do with the coding 
type python is using being different or something, Cannot remember or find 
it, it was quite a while back.
 years.
 Brian

***@***.***
Sent via blueyonder.
Please address personal E-mail to:-
***@***.***, putting 'Brian Gaff'
in the display name field.
Newsgroup monitored: alt.comp.blind-users
----- Original Message ----- 
From: "Sean Budd" ***@***.***>
To: "nvaccess/nvda" ***@***.***>
Cc: "Subscribed" ***@***.***>
Sent: Monday, January 24, 2022 12:53 AM
Subject: [nvaccess/nvda] NVDA installer does not start if special characters 
are in the installer path (Issue #13270)




### Steps to reproduce:

1. Place a copy of a NVDA installer (eg 2021.3.1) in a folder named `Testé ç 
ş ğ`
1. Run the installer
1. Observe popup dialog "NSIS Error" "Error launching installer"
1. Note that no nvda log is created, this is an early failure.

### Actual behavior:

NVDA installer does not start if special characters are in the installer 
path

### Expected behavior:

NVDA installer should start if special characters are in the installer path

### System configuration



#### NVDA installed/portable/running from source:

Installer

#### NVDA version:

2021.1 , 2021.1.3, alpha-24501,b45926cd

#### Windows version:

Windows 11 (10.0.22000 Build 22000)

#### Name and version of other software in use when reproducing the issue:

#### Other information about your system:

### Other questions
#### Does the issue still occur after restarting your computer?

#### Have you tried any other versions of NVDA? If so, please report their 
behaviors.

#### If NVDA add-ons are disabled, is your problem still occurring?

#### Does the issue still occur after you run the COM Registration Fixing 
Tool in NVDA's tools menu?


-- 
Reply to this email directly or view it on GitHub:
https://github.com/nvaccess/nvda/issues/13270

**Comment by CyrilleB79** [comment]
I have tested here with NVDA 2017.3, 2019.2.1, 2019.3.1, 2021.1, 2021.3.1 installers, launching the installer both when NVDA 2021.3.1 was running or when no instance of NVDA was running.
In all these cases, I can reproduce the issue.

I am using Windows 10 20H2 (64-bit) build 19042.1415, with French localization.
My codepage is CP1252:
```
prompt>REG QUERY "HKLM\SYSTEM\CurrentControlSet\Control\Nls\CodePage" -v ACP|find /I "ACP"
   ACP    REG_SZ    1252
```

Interestingly, if I rename the folder "Testé ç" to "Testé ç", the issue does not occur anymore. The two special characters é and ç are used in French and belong to the CP1252 codepage. On the contrary, ş and ğ special characters are note used in French and do not belong to CP1252 codepage.

Also, in case it matters, I have the following checkbox unchecked:
"Beta: Use Unicode UTF-8 for worldwide language support"

**Comment by lukaszgo1** [comment]
A good start for fixing this would be to upgrade to the latest version of NSIS which would resolve #9134 at the same time. Assuming that would not solve the issue a next reasonable thing to do would be to attempt to switch to [Unicode mode](https://nsis.sourceforge.io/Reference/Unicode). I've looked into this briefly in the past and the main problem which needs to be solved is that the plugin which were using for elevation during uninstallation is incompatible with the Unicode mode and since it is deprecated this is unlikely to change.

**Comment by ruifontes** [comment]
My first tests were performed with the NVDA language set to Portuguese(Portugal) and I didn't have any problem.
Today I have set NVDA language to Windows predefinition and I have the same problem of installer not loading...

**Comment by seanbudd** [comment]
I agree @lukaszgo1 that upgrading NSIS (and moving it to a submodule) would be a good thing to do, even if it doesn't resolve this issue. I suspect it's an NSIS issue so we might as well test the latest NSIS before logging an issue upstream.

**Comment by seanbudd** [comment]
I have tested with a copy of the NVDA installer built with NSIS 3.08, and this issue is fixed.
